### ISSUE TITLE
Fatal error when running Cron Job

### ISSUE THREAD

**Issue by LarsRezelman** [EmicoEcommerce/Magento2TweakwiseExport-archived#36]
When running the CronJob  emico_tweakwise_export we get a fatal error since the latests tweakwise export version 1.0.12.

Environment
--
- [ ] PHP Version: (|7.0)
- [ ] Magento Version: 2.1.10
- [ ] Magento Enterprise: (yes)
- [ ] Tweakwise Export Version: 1.0.12
- [ ] Magento Deploy Mode: (production)

![screen shot 2017-11-29 at 11 22 17](https://user-images.githubusercontent.com/32707547/33370881-36bdf928-d4f9-11e7-94f8-7898d73eb5fe.png)

**Comment by Fgruntjes** [comment]
Hello,

We just installed a clean Magento Enterprise 2.1.0 with sample data. Added the Tweakwise module to this installation and ran the export using the following commands.
```bash
magerun2 sys:cron:run emico_tweakwise_export
php bin/magento tweakwise:export
```

We discovered a bug with simple products that where used both in configurables as visible in the catalog itself. This bug has been fixed and a new release was made: V1.0.12, however this would not be the issue you describe here.

This issue looks a little bit like an issue with not clearing cache and or generated code properly. If you feel that this is a bug on our side please provide us with instruction to reproduce the issue based on a clean Magento installation with sample data.

Kind regards,
Freek

**Comment by Fgruntjes** [comment]
Does the issue still persist? So we can close the issue.

**Comment by LarsRezelman** [comment]
Hi Freek,

I do not understand your question. We did not update the Tweakwise extension since 1.0.13, so the issue still persists.

**Comment by Fgruntjes** [comment]
Hello,

Ok so the issue was not solved with the 1.0.13 version. I strongly have the feeling that this is not an issue with the module but with the shop you are using it in. It looks like something went wrong with code generation or not clearing cache correctly, however thats pure guesses.

If you are sure that this is not an issue with the shop you are using it in pleas provide steps to reproduce from a clean new Magento installation.

**Comment by LarsRezelman** [comment]
Hi Freek,

Did you have seen the comment form our developer Bassel al Araaj?

> at the moment when we run the cronjob emico_tweakwise_export, the product collection that is build in this export module give sometimes $storeid = NULL to this construct that causes the issue #36 , because NULL is not a integer.

Did you have checked to solve this issue?

**Comment by Fgruntjes** [comment]
@LarsRezelman 

No I did not see any coments of "Bassel al Araaj" It would be usefull if he would comment on the issue in github so all the information is on one place.

Also please read my previous comment:
> If you are sure that this is not an issue with the shop you are using it in pleas provide steps to reproduce from a clean new Magento installation.

Currently we have a screenshot of half an exception stack trace and a description that does not reproduce the issue on a default Magento installation. For example how was the cronjob even started in the first place? (magerun?) You really need to provide us with enough information to reproduce the issue from a clean Magento installation. Otherwise there is no way for us to fix this issue.

We expect a step by step procedure on reproducing the issue from a new Magento installation for example:

1. Install Magento EE 2.1 
```bash
composer create-project --no-interaction --ignore-platform-reqs --repository-url=https://repo.magento.com/ ${MAGE2_VERSION} build

php bin/magento setup:install --no-interaction \
    --admin-email "$MAGE2_ADMIN_EMAIL" \
    --admin-firstname "$MAGE2_ADMIN_FIRST_NAME" \
    --admin-lastname "$MAGE2_ADMIN_LAST_NAME" \
    --admin-password "$MAGE2_ADMIN_PASSWORD" \
    --admin-user "$MAGE2_ADMIN_USERNAME" \
    --backend-frontname admin \
    --base-url "$MAGE2_FAKE_URL" \
    --db-host "$MAGE2_DB_HOST" \
    --db-name "$MAGE2_DB_NAME" \
    --db-user "$MAGE2_DB_USER" \
    --db-password "$MAGE2_DB_PASSWORD" \
    --session-save files \
    --use-rewrites 1 \
    --use-secure 0 \
    -vvv
```

2. Install example data
```bash
php bin/magento sampledata:deploy
```
3. Install TW Export module
```bash
composer require emico/tweakwise-export
php bin/magento module:enable Emico_TweakwiseExport
php bin/magento setup:upgrade
```

4. Trigger export 
```bash
magerun2 sys:cron:run emico_tweakwise_export
```

If this not possible please create a pull request with a proposed fix for the issue so we can review and add it to the code base.

**Comment by joost-florijn-kega** [comment]
@Fgruntjes 

I have looked in to the issue and i came across the issue that when Magento has more then 1000 products of any type (bundled, simple, vitrual, configurable, etc) the new product collection isn't initialised probably. 

The reason why the cron works with the sample data of Magento is that the sample data doesn't contain more then 1000 products.

I have created an pull request the fix the problem.
